src/functions/getFirstExistingParentPath.ts

```diff
@@ -8,15 +8,19 @@ import Dependencies from '@/src/types/dependencies'
  * @param dependencies - Dependencies container
  */
 async function getFirstExistingParentPath(directoryPath: string, dependencies: Dependencies): Promise<string> {
-	let parentDirectoryPath = directoryPath
+	let parentDirectoryPath = dependencies.pathNormalize(directoryPath)
 	let parentDirectoryFound = await isDirectoryExisting(parentDirectoryPath, dependencies)
 
-	while (!parentDirectoryFound) {
-		parentDirectoryPath = dependencies.pathNormalize(parentDirectoryPath + '/..')
+	for (let i = 0; i < 4096 && !parentDirectoryFound; ++i) {
```

Comment:
I'm curious why the loop limit is `4096`. Can we add a variable for it?

Reply:
ah yeah. figured I'd get a question about this. I should have added a comment. 

While loops are error prone (and the reason we forked in the first place), so I just wanted to set a theoretical limit on the number of folders we'd have to traverse and convert to a for loop. I think the max number is < 4096 though. I'll update it and put it in a variable with a comment

Reply:
okay updated!